src/controller.rs

```diff
@@ -498,6 +542,31 @@ fn pod_template(config: &WasmCloudHostConfig, _ctx: Arc<Context>) -> PodTemplate
     template
 }
 
+async fn discover_configmap_certificates(
+    namespace: String,
+    name: String,
+    ctx: &Context,
+) -> Result<Vec<String>> {
```

Comment:
Generally the rule of thumb we should probably move towards is to take `&str` in function arguments, and return `String` as a result, so this would be changed slightly to:
```suggestion
async fn discover_configmap_certificates(
    namespace: &str,
    name: &str,
    ctx: &Context,
) -> Result<Vec<String>> {
```

Reply:
For more context, I think [this article is a good read on the "why"](https://dx13.co.uk/articles/2024/02/24/rust-string-and-str/).